engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -641,7 +642,8 @@ private void preRenderUpdate(RenderingStage renderingStage) {
         renderableWorld.queueVisibleChunks(isFirstRenderingStageForCurrentFrame);
 
         if (requestedTaskListRefresh) {
-            renderTaskListGenerator.refresh();
+            List<Node> orderedNodes = renderGraph.getNodesInTopologicalOrder();
```

Comment:
Another problematic area in the past. `refresh()` basically uses the exact same Nodes as last time. Which means even though if we update the renderGraph, `refresh()` won't actually include the new Nodes in the final taskList. Since we were dealing with only 42 Nodes till now (with some being enabled/disabled), it did not matter. Now that we're dealing with 42 plus/minus some Nodes, we have to generate the taskList from the new set of Nodes (which might or might not be the same as last time).

Reply:
I noticed the change to the enabling/disabling of conditional nodes. The explanation above is not clear to me. Can you please rephrase everything in a new explanation?